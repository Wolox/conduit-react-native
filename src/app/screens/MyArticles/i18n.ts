import i18next from 'i18next';

i18next.addResources('en', 'MY_ARTICLES', {
  TITLE: 'My articles',
  NO_ARTICLES: 'No articles are here... yet'
});
