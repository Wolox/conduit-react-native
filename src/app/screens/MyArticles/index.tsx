import React, { useCallback, useEffect } from 'react';
import { FlatList, ListRenderItem, SafeAreaView, View } from 'react-native';
import { useDispatch, useSelector } from 'react-redux';
import i18next from 'i18next';
import ScreenWithLoader from '@components/ScreenWithLoader';
import CustomText from '@components/CustomText';
import useNavigation from '@components/AppNavigator/helper';
import ArticleItem from '@components/ArticleItem';
import Routes from '@constants/routes';
import { MyArticlesState, State } from '@interfaces/reduxInterfaces';
import { Article } from '@interfaces/articlesInterface';
import { ListKeyExtractor } from '@interfaces/miscelanious';
import MyActiclesActions from '@redux/myArticles/actions';
import { UserResponse } from '@interfaces/authInterfaces';
import { Nullable } from '@interfaces/globalInterfaces';

import './i18n';
import styles from './styles';
import Header from './Header';

export default function MyArticles() {
  const navigation = useNavigation();
  const {
    myArticles: { articles },
    myArticlesLoading
  } = useSelector<State, MyArticlesState>(state => state.myArticles);
  const currentUser = useSelector<State, Nullable<UserResponse>>(state => state.auth.currentUser);
  const renderSeparator = useCallback(() => <View style={styles.separator} />, []);
  const handlePressArticle = useCallback(
    (article: Article) => navigation?.navigate(Routes.DetailArticle, { article }),
    [navigation]
  );
  const renderItem: ListRenderItem<Article> = useCallback(
    ({ item }) => <ArticleItem item={item} onPress={handlePressArticle} />,
    [handlePressArticle]
  );
  const keyExtractor: ListKeyExtractor<Article> = useCallback(({ slug }) => `${slug}`, []);
  const renderMessage = useCallback(
    () => (
      <>
        {articles?.length ? (
          <FlatList<Article>
            data={articles}
            keyExtractor={keyExtractor}
            renderItem={renderItem}
            ItemSeparatorComponent={renderSeparator}
            showsVerticalScrollIndicator={false}
          />
        ) : (
          <View style={styles.noItemMessage}>
            <CustomText>{i18next.t('MY_ARTICLES:NO_ARTICLES')}</CustomText>
          </View>
        )}
      </>
    ),
    [keyExtractor, articles, renderItem, renderSeparator]
  );
  const dispatch = useDispatch();
  useEffect(() => {
    if (currentUser) {
      dispatch(MyActiclesActions.getMyArticles(currentUser));
    }
  }, [currentUser, dispatch]);

  return (
    <SafeAreaView style={styles.container}>
      <Header />
      <ScreenWithLoader loading={myArticlesLoading} withInitialLoading={false}>
        {renderMessage()}
      </ScreenWithLoader>
    </SafeAreaView>
  );
}
