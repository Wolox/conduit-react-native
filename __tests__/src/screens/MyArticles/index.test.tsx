import React from 'react';
import renderer from 'react-test-renderer';
import MockedNavigator from '@mocks/MockedNavigator';
import MyArticles from '@screens/MyArticles';

jest.mock('react-redux', () => {
  return {
    useSelector: jest
      .fn()
      .mockImplementation(() => ({ myArticles: { articles: null }, myArticlesLoading: false })),
    useDispatch: () => jest.fn()
  };
});

describe('test MyArticles_SCREEN', () => {
  test('to match with snapshot', async () => {
    const tree = await renderer.create(<MockedNavigator component={MyArticles} />).toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree).not.toBeNull();
  });
});
