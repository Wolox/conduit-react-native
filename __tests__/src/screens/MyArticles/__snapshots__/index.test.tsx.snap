// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test MyArticles_SCREEN to match with snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      pointerEvents="box-none"
      style={
        Object {
          "zIndex": 1,
        }
      }
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        onLayout={[Function]}
        pointerEvents="box-none"
        style={null}
      >
        <View
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "zIndex": 0,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "rgb(255, 255, 255)",
                "borderBottomColor": "rgb(216, 216, 216)",
                "flex": 1,
                "shadowColor": "rgb(216, 216, 216)",
                "shadowOffset": Object {
                  "height": 0.5,
                  "width": 0,
                },
                "shadowOpacity": 0.85,
                "shadowRadius": 0,
              }
            }
          />
        </View>
        <View
          pointerEvents="box-none"
          style={
            Object {
              "height": 64,
              "maxHeight": undefined,
              "minHeight": undefined,
              "opacity": undefined,
              "transform": undefined,
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Object {
                "height": 20,
              }
            }
          />
          <View
            pointerEvents="box-none"
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "marginHorizontal": 16,
                  "opacity": 1,
                }
              }
            >
              <Text
                accessibilityRole="header"
                aria-level="1"
                numberOfLines={1}
                onLayout={[Function]}
                style={
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 17,
                    "fontWeight": "600",
                  }
                }
              >
                MockedScreen
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <RNSScreenContainer
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <RNSScreen
        activityState={2}
        forwardedRef={[Function]}
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "opacity": 1,
            }
          }
        />
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onGestureEnd={[Function]}
          onOpen={[Function]}
          onTransition={[Function]}
          pointerEvents="box-none"
          style={
            Array [
              Object {
                "overflow": undefined,
              },
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
            ]
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            needsOffscreenAlphaCompositing={false}
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              collapsable={false}
              forwardedRef={[Function]}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              style={
                Object {
                  "flex": 1,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "backgroundColor": "rgb(242, 242, 242)",
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": -1,
                    },
                    "shadowOpacity": 0.3,
                    "shadowRadius": 5,
                    "top": 0,
                    "width": 3,
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "overflow": "hidden",
                    },
                    Array [
                      Object {
                        "backgroundColor": "rgb(242, 242, 242)",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "flexDirection": "column-reverse",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <RCTSafeAreaView
                      emulateUnlessSupported={true}
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "marginTop": 2,
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "borderBottomColor": "#5CB85C",
                              "borderBottomWidth": 2,
                            }
                          }
                        >
                          <Text
                            style={
                              Array [
                                Object {
                                  "backgroundColor": "transparent",
                                  "color": "#000",
                                  "fontFamily": "Nunito",
                                  "fontSize": 25.142857142857142,
                                  "fontStyle": "normal",
                                  "fontWeight": "400",
                                },
                                Array [],
                                undefined,
                              ]
                            }
                          >
                            MY_ARTICLES:TITLE
                          </Text>
                        </View>
                      </View>
                      <Modal
                        animationType="fade"
                        hardwareAccelerated={false}
                        transparent={true}
                        visible={false}
                      >
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "#00000033",
                              "flex": 1,
                              "justifyContent": "center",
                              "marginTop": 22,
                            }
                          }
                        >
                          <View
                            style={
                              Object {
                                "alignItems": "center",
                                "backgroundColor": "#FFF",
                                "borderRadius": 20,
                                "elevation": 5,
                                "height": 180,
                                "margin": 20,
                                "padding": 35,
                                "shadowColor": "#555555",
                                "shadowOffset": Object {
                                  "height": 2,
                                  "width": 0,
                                },
                                "shadowOpacity": 0.3,
                                "shadowRadius": 4,
                                "width": "93%",
                              }
                            }
                          >
                            <Text
                              style={
                                Array [
                                  Object {
                                    "backgroundColor": "transparent",
                                    "color": "#000",
                                    "fontFamily": "Nunito",
                                    "fontSize": 25.142857142857142,
                                    "fontStyle": "normal",
                                    "fontWeight": "400",
                                  },
                                  Array [
                                    Object {
                                      "color": "#000",
                                      "fontFamily": "Nunito",
                                      "fontSize": 25.142857142857142,
                                      "fontStyle": "normal",
                                      "fontWeight": "700",
                                    },
                                  ],
                                  Object {
                                    "marginBottom": 15,
                                    "textAlign": "center",
                                  },
                                ]
                              }
                            >
                              HOME:CONFIRM_DELETE_ARTICLE
                            </Text>
                            <View
                              style={
                                Object {
                                  "flexDirection": "row",
                                  "justifyContent": "space-evenly",
                                  "marginTop": 20,
                                  "width": "100%",
                                }
                              }
                            >
                              <View
                                accessible={true}
                                collapsable={false}
                                focusable={true}
                                onBlur={[Function]}
                                onClick={[Function]}
                                onFocus={[Function]}
                                onResponderGrant={[Function]}
                                onResponderMove={[Function]}
                                onResponderRelease={[Function]}
                                onResponderTerminate={[Function]}
                                onResponderTerminationRequest={[Function]}
                                onStartShouldSetResponder={[Function]}
                                style={
                                  Array [
                                    Object {
                                      "borderRadius": 20,
                                      "elevation": 2,
                                      "padding": 15,
                                    },
                                    Object {
                                      "backgroundColor": "#5CB85C",
                                    },
                                  ]
                                }
                              >
                                <Text
                                  style={
                                    Object {
                                      "color": "#FFF",
                                      "fontWeight": "bold",
                                      "textAlign": "center",
                                    }
                                  }
                                >
                                  CONFIRMATION_MODAL:CANCEL
                                </Text>
                              </View>
                              <View
                                accessible={true}
                                collapsable={false}
                                focusable={true}
                                onBlur={[Function]}
                                onClick={[Function]}
                                onFocus={[Function]}
                                onResponderGrant={[Function]}
                                onResponderMove={[Function]}
                                onResponderRelease={[Function]}
                                onResponderTerminate={[Function]}
                                onResponderTerminationRequest={[Function]}
                                onStartShouldSetResponder={[Function]}
                                style={
                                  Array [
                                    Object {
                                      "borderRadius": 20,
                                      "elevation": 2,
                                      "padding": 15,
                                    },
                                    Object {
                                      "backgroundColor": "#FF2400",
                                    },
                                  ]
                                }
                              >
                                <Text
                                  style={
                                    Object {
                                      "color": "#FFF",
                                      "fontWeight": "bold",
                                      "textAlign": "center",
                                    }
                                  }
                                >
                                  CONFIRMATION_MODAL:DELETE
                                </Text>
                              </View>
                            </View>
                          </View>
                        </View>
                      </Modal>
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "flex": 1,
                            "marginTop": "10%",
                          }
                        }
                      >
                        <Text
                          style={
                            Array [
                              Object {
                                "backgroundColor": "transparent",
                                "color": "#000",
                                "fontFamily": "Nunito",
                                "fontSize": 25.142857142857142,
                                "fontStyle": "normal",
                                "fontWeight": "400",
                              },
                              Array [],
                              undefined,
                            ]
                          }
                        >
                          MY_ARTICLES:NO_ARTICLES
                        </Text>
                      </View>
                    </RCTSafeAreaView>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RNSScreen>
    </RNSScreenContainer>
  </RNCSafeAreaProvider>
</View>
`;
